ide-support/lsp/lsp-server/src/main/java/io/helidon/lsp/server/service/config/yaml/YamlParser.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.lsp.server.service.config.yaml;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Parser for the files in YAML format.
+ */
+class YamlParser {
```

Comment:
If we are implementing a YAML parser from scratch it would be good to have it under common (maybe as a new module similar to `common/xml`, maybe `common/yaml`).

Reply:
I am not sure that it is a fully functional yaml parser and it can be used somewhere else. The main goal why I decide to create it that it has to be tolerant to errors in an yaml document. It only recognizes keys and their places in the document.

Reply:
Roger that.